File: SDKs/github.com/ghes-3.1/1.1.4/python/src/sdk/models/operations/repos_create_deployment_status.py
from dataclasses import dataclass, field
from typing import Enum,List,Optional
from dataclasses_json import dataclass_json
from sdk.models import shared


@dataclass
class ReposCreateDeploymentStatusPathParams:
    deployment_id: int = field(default=None, metadata={'path_param': { 'field_name': 'deployment_id', 'style': 'simple', 'explode': False }})
    owner: str = field(default=None, metadata={'path_param': { 'field_name': 'owner', 'style': 'simple', 'explode': False }})
    repo: str = field(default=None, metadata={'path_param': { 'field_name': 'repo', 'style': 'simple', 'explode': False }})
    
class ReposCreateDeploymentStatusRequestBodyEnvironmentEnum(str, Enum):
    PRODUCTION = "production"
    STAGING = "staging"
    QA = "qa"

class ReposCreateDeploymentStatusRequestBodyStateEnum(str, Enum):
    ERROR = "error"
    FAILURE = "failure"
    INACTIVE = "inactive"
    IN_PROGRESS = "in_progress"
    QUEUED = "queued"
    PENDING = "pending"
    SUCCESS = "success"


@dataclass_json
@dataclass
class ReposCreateDeploymentStatusRequestBody:
    auto_inactive: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'auto_inactive' }})
    description: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'description' }})
    environment: Optional[ReposCreateDeploymentStatusRequestBodyEnvironmentEnum] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'environment' }})
    environment_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'environment_url' }})
    log_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'log_url' }})
    state: ReposCreateDeploymentStatusRequestBodyStateEnum = field(default=None, metadata={'dataclasses_json': { 'field_name': 'state' }})
    target_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'target_url' }})
    

@dataclass
class ReposCreateDeploymentStatusRequest:
    path_params: ReposCreateDeploymentStatusPathParams = field(default=None)
    request: Optional[ReposCreateDeploymentStatusRequestBody] = field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class ReposCreateDeploymentStatusResponse:
    content_type: str = field(default=None)
    headers: dict[str, List[str]] = field(default=None)
    status_code: int = field(default=None)
    deployment_status: Optional[shared.DeploymentStatus] = field(default=None)
    validation_error: Optional[shared.ValidationError] = field(default=None)
    

File: SDKs/github.com/ghes-2.19/1.1.4/go/pkg/models/operations/reposcreatedeploymentstatus.go
package operations

import (
"openapi/pkg/models/shared")

type ReposCreateDeploymentStatusPathParams struct {
    DeploymentID int64 `pathParam:"style=simple,explode=false,name=deployment_id"`
    Owner string `pathParam:"style=simple,explode=false,name=owner"`
    Repo string `pathParam:"style=simple,explode=false,name=repo"`
    
}


type ReposCreateDeploymentStatusRequestBodyEnvironmentEnum string

const (
    ReposCreateDeploymentStatusRequestBodyEnvironmentEnumProduction ReposCreateDeploymentStatusRequestBodyEnvironmentEnum = "production"
ReposCreateDeploymentStatusRequestBodyEnvironmentEnumStaging ReposCreateDeploymentStatusRequestBodyEnvironmentEnum = "staging"
ReposCreateDeploymentStatusRequestBodyEnvironmentEnumQa ReposCreateDeploymentStatusRequestBodyEnvironmentEnum = "qa"
)



type ReposCreateDeploymentStatusRequestBodyStateEnum string

const (
    ReposCreateDeploymentStatusRequestBodyStateEnumError ReposCreateDeploymentStatusRequestBodyStateEnum = "error"
ReposCreateDeploymentStatusRequestBodyStateEnumFailure ReposCreateDeploymentStatusRequestBodyStateEnum = "failure"
ReposCreateDeploymentStatusRequestBodyStateEnumInactive ReposCreateDeploymentStatusRequestBodyStateEnum = "inactive"
ReposCreateDeploymentStatusRequestBodyStateEnumInProgress ReposCreateDeploymentStatusRequestBodyStateEnum = "in_progress"
ReposCreateDeploymentStatusRequestBodyStateEnumQueued ReposCreateDeploymentStatusRequestBodyStateEnum = "queued"
ReposCreateDeploymentStatusRequestBodyStateEnumPending ReposCreateDeploymentStatusRequestBodyStateEnum = "pending"
ReposCreateDeploymentStatusRequestBodyStateEnumSuccess ReposCreateDeploymentStatusRequestBodyStateEnum = "success"
)


type ReposCreateDeploymentStatusRequestBody struct {
    AutoInactive *bool `json:"auto_inactive,omitempty"`
    Description *string `json:"description,omitempty"`
    Environment *ReposCreateDeploymentStatusRequestBodyEnvironmentEnum `json:"environment,omitempty"`
    EnvironmentURL *string `json:"environment_url,omitempty"`
    LogURL *string `json:"log_url,omitempty"`
    State ReposCreateDeploymentStatusRequestBodyStateEnum `json:"state"`
    TargetURL *string `json:"target_url,omitempty"`
    
}

type ReposCreateDeploymentStatusRequest struct {
    PathParams ReposCreateDeploymentStatusPathParams 
    Request *ReposCreateDeploymentStatusRequestBody `request:"mediaType=application/json"`
    
}

type ReposCreateDeploymentStatusResponse struct {
    ContentType string 
    Headers map[string][]string 
    StatusCode int64 
    DeploymentStatus *shared.DeploymentStatus 
    ValidationError *shared.ValidationError 
    
}

